core/trino-main/src/test/java/io/trino/operator/output/TestSlicePositionsAppender.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.output;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.RunLengthEncodedBlock;
+import io.trino.spi.block.VariableWidthBlock;
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import org.testng.annotations.Test;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Optional;
+
+import static io.trino.block.BlockAssertions.assertBlockEquals;
+import static io.trino.block.BlockAssertions.createStringsBlock;
+import static io.trino.operator.output.SlicePositionsAppender.duplicateBytes;
+import static io.trino.spi.block.PageBuilderStatus.DEFAULT_MAX_PAGE_SIZE_IN_BYTES;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.nio.ByteOrder.nativeOrder;
+import static org.testng.internal.junit.ArrayAsserts.assertArrayEquals;
+
+public class TestSlicePositionsAppender
+{
+    @Test
+    public void testAppendEmptySliceRle()
```

Comment:
why this can't be part of `TestPositionsAppender`, e.g: `testAppendEmptyRle` (for given type)

Reply:
it was part of the `TestPositionsAppender` but not per type. Empty string/slice is only valid for `VARCHAR`.